fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersionsIT.java

```diff
@@ -183,6 +195,55 @@ public void testVersioningANodeWithAVersionableChild() throws IOException {
     }
 
     @Test
+    public void testVersionHeaders() throws IOException {
+        final String id = getRandomUniqueId();
+        createObjectAndClose(id);
+        enableVersioning(id);
+
+        createDatastream(id, "ds", "This DS will be versioned");
+        final String dsId = id + "/ds";
+        enableVersioning(dsId);
+
+        // Version datastream
+        postObjectVersion(dsId, "label");
+
+        final String versionId = dsId + "/fcr:versions/label";
+
+        final Link NON_RDF_SOURCE_LINK = fromUri(NON_RDF_SOURCE.getURI()).rel(TYPE.getLocalName()).build();
+
+        final Link DESCRIBED_BY_LINK = fromUri(serverAddress + versionId + "/" + FCR_METADATA).rel(
+                DESCRIBED_BY.getLocalName()).build();
+
+        // Look for expected Link headers
+        final HttpHead headObjMethod = headObjMethod(versionId);
+        try (final CloseableHttpResponse response = execute(headObjMethod)) {
+
+            final Set<Link> resultSet = new HashSet<>();
+            final Collection<String> linkHeaders = getLinkHeaders(response);
+
+            linkHeaders.stream().map(Link::valueOf).forEach(link -> {
```

Comment:
rather than a `forEach` adding to a `Set<Link>` here, a `map` and `collect(toSet())` would be cleaner.